README.md

```diff
@@ -1,64 +1,134 @@
+`ssh` into the master node and run `spark-shell` as above for an interactive
+environment, or use `spark-submit` for batch jobs.  (You may submit Steps to
+the EMR cluster using `spark-submit` as well.)
 
-### Collating Feature Groups into a VectorTile
+### Importing Data ###
 
-Once clipped and gridded by `VectorPipe.toGrid`, we have a `RDD[(SpatialKey,
-Iterable[Feature[G, D]])]` that represents all the Geometry fragments
-present at each tiled location on the earth. This is the perfect shape to
-turn into a `VectorTile`. To do so, we need to choose a *Collator* function,
-which determines what VectorTile Layer each `Feature` should be placed into,
-and how (if at all) its corresponding metadata (the `D`) should be
-processed.
+Batch analysis can be performed in a few different ways.  Perhaps the fastest
+way is to procure an OSM PBF file from a source such as
+[GeoFabrik](https://download.geofabrik.de/index.html), which supplies various
+extracts of OSM, including the full planet worth of data.
 
-Want to write your own Collator? The `Collate.generically` function will be
-of interest to you.
+VectorPipe does not provide the means to directly read these OSM PBF files,
+however, and a conversion to a useful file format will thus be needed.  We
+suggest using [`osm2orc`](https://github.com/mojodna/osm2orc) to convert your
+source file to the ORC format which can be read natively via Spark:
+```scala
+val df = spark.read.orc(path)
+```
+The resulting `DataFrame` can be processed with VectorPipe.
+
+It is also possible to read from a cache of
+[OsmChange](https://wiki.openstreetmap.org/wiki/OsmChange) files directly
+rather than convert the PBF file:
+```scala
+import vectorpipe.sources.Source
+val df = spark.read
+              .format(Source.Changes)
+              .options(Map[String, String](
+                Source.BaseURI -> "https://download.geofabrik.de/europe/isle-of-man-updates/",
+                Source.StartSequence -> "2080",
+                Source.EndSequence -> "2174",
+                Source.BatchSize -> "1"))
+              .load
+              .persist // recommended to avoid rereading
+```
+This may issue errors (including reads timing out), but should complete.  This
+is much slower than using ORC files and is much touchier, but it stands as an
+option.
+
+[It is also possible to build a dataframe from a stream of changesets in a
+similar manner as above.  Changesets carry additional metadata regarding the
+author of the changes, but none of the geometric information.  These tables
+can be joined on `changeset`.]
+
+In either case, a useful place to start is to convert the incoming dataframe
+into a more usable format.  We recommend calling
+```scala
+val geoms = ProcessOSM.constructGeometries(df)
+```
+which will produce a frame consisting of "top-level" entities, which is to say
+nodes that don't participate in a way, ways that don't participate in
+relations, and a subset of the relations from the OSM data.  The resulting
```

Comment:
It would be great to get an explicit list of the entities that will be produced by `constructGeometries`.  @mojodna ?

Reply:
* points (from tagged nodes, including tags that really ought to be dropped, e.g. `source=*`)
* polygons (from ways with tags that are cause them to be considered as areas)
* lines (from ways without area tags)
* multipolygons (from multipolygon or boundary relations)
* multilinestrings (from route relations)

This currently does include ways that participate in relations (as long as they have tags? would need to check that. ways with tagging that's distinct from the relation suggests that it fulfills a separate, distinct role) and tagged nodes that participate in ways and relations (same deal as above; a bollard may contribute a vertex to a path).

Reply:
(I want to revisit the relation assembly process (for non-multipolygons) to prevent new geometries from being created and just propagating relation tags onto the constituent lines / polygons so that they can fulfill the same role without duplication. It does complicate things a bit, resulting in more minor versions of ways, as changes to relations that they're part of will change their metadata (similar to how changes to nodes change their geometry).)